Intersect.Server.Core/Entities/Entity.cs

```diff
@@ -3056,14 +3059,17 @@ protected virtual void DropItems(Entity killer, bool sendUpdate = true)
             // Spawn the actual item!
             if (MapController.TryGetInstanceFromMap(MapId, MapInstanceId, out var instance))
             {
-                instance.SpawnItem(X, Y, drop, drop.Quantity, lootOwner, sendUpdate);
+                var itemSource = this.GetItemSource();
+                instance.SpawnItem(itemSource, X, Y, drop, drop.Quantity, lootOwner, sendUpdate);
             }
 
             // Process the drop (for players this would remove it from their inventory)
             OnDropItem(slot, drop);
         }
     }
 
+    protected abstract EntityItemSource GetItemSource();
```

Comment:
This should be called `AsItemSource()`

The naming rationale is that `Get` is trying to fetch a copy of something the object has, but the item source is a reference, not a property of the entity (which would just be a property in cases where there's no parameter).